inc/class-s3-uploads.php

```diff
@@ -208,12 +209,42 @@ public function filter_editors( $editors ) {
 	 */
 	public function filter_sideload_move_temp_file_to_s3( array $file ) {
 		$upload_dir = wp_upload_dir();
-		$new_path = $upload_dir['basedir'] . '/tmp/' . basename( $file['tmp_name'] );
+		$new_path   = $upload_dir['basedir'] . '/tmp/' . basename( $file['tmp_name'] );
 
 		copy( $file['tmp_name'], $new_path );
 		unlink( $file['tmp_name'] );
 		$file['tmp_name'] = $new_path;
 
 		return $file;
 	}
+
+	/**
+	 * Filters wp_read_image_metadata. exif_read_data() doesn't work on
+	 * file streams so we need to make a temporary local copy to extract
+	 * exif data from.
+	 *
+	 * @param array  $meta
+	 * @param string $file
+	 * @return array|bool
+	 */
+	public function wp_filter_read_image_metadata( $meta, $file ) {
+		remove_filter( 'wp_read_image_metadata', array( $this, 'wp_filter_read_image_metadata' ), 10, 2 );
+		$temp_file = $this->copy_image_from_s3( $file );
```

Comment:
You seem to be missing the `callback` param, also see other warnings in https://travis-ci.org/humanmade/S3-Uploads/jobs/150781407